cli/functionary/tokens.py

```diff
@@ -1,8 +1,28 @@
-from .client import post
+import json
 
+import click
+import requests
 
-def login(user: str, password: str):
-    login_response = post(
-        "api-token-auth", data={"username": user, "password": password}
-    )
-    return login_response.get("token")
+
+def login(login_url, user: str, password: str):
+    try:
+        login_response = requests.post(
+            f"{login_url}", data={"username": user, "password": password}
+        )
+        # check status code/message on return then exit
+        if login_response.ok:
+            token = json.loads(login_response.text).get("token")
+            return token
+        elif login_response.status_code == 400:
+            raise click.ClickException(
+                "Code 400: Unable to login due to an invalid username or password."
```

Comment:
You don't need the "Code 400" part here.  That's surfacing too much of the implementation detail to the user.  Just the "Unable to login..." bit is necessary.